File: project/hyeok-onboarding/src/main/java/com/INNER_CIRCLE_ICD4/innerCircle/survey/request/domain/Answer.java
package com.INNER_CIRCLE_ICD4.innerCircle.survey.request.domain;

import jakarta.persistence.*;

@Entity
public class Answer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "response_id")
    private Response response;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id")
    private Question question;

    @Column(columnDefinition = "TEXT")
    private String answerValue;  // 사용자 입력값 또는 선택된 옵션값

    // === Getters/Setters ===
    public Long getId() {
        return id;
    }

    public Response getResponse() {
        return response;
    }

    public void setResponse(Response response) {
        this.response = response;
    }

    public Question getQuestion() {
        return question;
    }

    public void setQuestion(Question question) {
        this.question = question;
    }

    public String getAnswerValue() {
        return answerValue;
    }

    public void setAnswerValue(String answerValue) {
        this.answerValue = answerValue;
    }
}

File: project/sanghyuk-onboarding/src/main/kotlin/fc/innercircle/sanghyukonboarding/formreply/model/Answer.kt
package fc.innercircle.sanghyukonboarding.formreply.model

import fc.innercircle.sanghyukonboarding.common.domain.model.BaseEntity
import jakarta.persistence.ConstraintMode
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.ForeignKey
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
open class Answer(
    formReply: FormReply,
    val questionSnapshotId: String,
    // 주관식용 답변
    val answer: String = "",
    // 선택형 답변용 선택 옵션 ID
    val selectableOptionId: String = "",
    createdBy: String,
) : BaseEntity(createdBy) {

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(
        name = "form_reply_id",
        nullable = false,
        foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT),
        columnDefinition = "char(26) not null comment '설문 응답 ID'"
    )
    var formReply: FormReply = formReply
        protected set
}
